components/suggest/src/rs.rs

```diff
@@ -100,30 +100,44 @@ pub(crate) enum SuggestRecord {
     GlobalConfig(DownloadedGlobalConfig),
 }
 
+/// Implemented by the various attachment types
+pub trait SuggestAttachment: DeserializeOwned + std::fmt::Debug {
+    /// Get fields that we saw when ingesting this attachment, but don't recognize
+    fn unknown_fields(&self) -> &serde_json::Map<String, serde_json::Value>;
+
+    fn has_unknown_fields(&self) -> bool {
+        !self.unknown_fields().is_empty()
+    }
+}
+
 /// Represents either a single value, or a list of values. This is used to
 /// deserialize downloaded attachments.
 #[derive(Clone, Debug, Deserialize)]
 #[serde(untagged)]
-enum OneOrMany<T> {
+pub enum OneOrMany<T> {
```

Comment:
Suggestion: how about renaming this to something more readable as it's now public and `SuggestAttachment<T>` has been re-purposed as a trait.

`OneOrMany` was used to model the payload for each RS attachment, which can be either a list of JSON values or a single JSON value. Maybe call it `AttachmentPayload`.

WDYT?